sympy/vector/coordsysrect.py

```diff
@@ -45,10 +50,13 @@ def __new__(cls, name, location=None, rotation_matrix=None,
 
         """
 
+        global vector_name_list, variable_name_list, vector_alias, variable_alias
```

Comment:
Have a look at `sympy.diffgeom`, that code is well written. If possible, I would suggest to take the `sympy.diffgeom.CoordSystem` class as a reference:

http://docs.sympy.org/latest/modules/diffgeom.html#sympy.diffgeom.CoordSystem